public/app/plugins/datasource/stackdriver/datasource.ts

```diff
@@ -287,7 +288,7 @@ export default class StackdriverDatasource extends DataSourceApi<StackdriverQuer
 
       return this.metricTypes;
     } catch (error) {
-      appEvents.emit('ds-request-error', this.formatStackdriverError(error));
+      appEvents.emit(dsRequestError, { error: { data: { error: this.formatStackdriverError(error) } } });
```

Comment:
Just noticed this one, out of curiosity why this change? Why not add this `error` to the `this. formatStackdriverError `?

Reply:
Made this change since the handler for `dsRequestError` events (`QueryInspector > onDataSourceResponse`) doesn't deal with strings (which was previously being sent), so I changed it so that it's sent an object it can properly deal with. I didn't change what `formatStackdriverError` returns since elsewhere where it's used it _does_ expect a string. Not a replacement for proper typing, but it's a quick fix until we have well-defined error types.

Reply:
Ok, thanks for clarification